harness/determined/cli/token.py

```diff
@@ -0,0 +1,193 @@
+
+
+def revoke_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    try:
+        request = bindings.v1PatchAccessTokenRequest(
+            tokenId=args.token_id, description=None, setRevoked=True
+        )
+        resp = bindings.patch_PatchAccessToken(sess, body=request, tokenId=args.token_id)
+        print(json.dumps(resp.to_json(), indent=2))
+    except api.errors.NotFoundException:
+        raise errors.CliError("Token not found")
+    print(f"Successfully updated token with ID: {args.token_id}.")
+
+
+def create_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    username = args.username or sess.username
+    user = bindings.get_GetUserByUsername(session=sess, username=username).user
+
+    if user is None or user.id is None:
+        raise errors.CliError(f"User '{username}' not found or does not have an ID")
+
+    request = None
+    request = bindings.v1PostAccessTokenRequest(
+        userId=user.id, lifespan=args.expiration_duration, description=args.description
+    )
+    resp = bindings.post_PostAccessToken(sess, body=request).to_json()
+
+    output_string = None
+    if args.yaml:
+        output_string = util.yaml_safe_dump(resp, default_flow_style=False)
+    elif args.json:
+        output_string = json.dumps(resp, indent=2)
+    else:
+        output_string = f'{resp["token"]}\n{resp["tokenId"]}'
+
+    if args.output_file:
+        with open(args.output_file, "w") as file:
+            file.write(output_string)
+    else:
+        print(output_string)
+
+
+def edit_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    try:
+        if args.description and args.token_id:
+            request = bindings.v1PatchAccessTokenRequest(
+                tokenId=args.token_id, description=args.description, setRevoked=False
+            )
+            resp = bindings.patch_PatchAccessToken(sess, body=request, tokenId=args.token_id)
+        print(json.dumps(resp.to_json(), indent=2))
+    except api.errors.APIException as e:
+        raise errors.CliError(f"Caught APIException: {str(e)}")
+    except api.errors.NotFoundException:
+        raise errors.CliError("Token not found")
+    print(f"Successfully updated token with ID: {args.token_id}.")
+
+
+def login_with_token(args: argparse.Namespace) -> None:
+    unauth_session = api.UnauthSession(master=args.master, cert=cli.cert)
+    auth_headers = {"Authorization": f"Bearer {args.token}"}
+    user_data = unauth_session.get("/api/v1/me", headers=auth_headers).json()
+    if "user" in user_data and "username" in user_data.get("user"):
+        username = user_data.get("user").get("username")
+
+    token_store = authentication.TokenStore(args.master)
+    token_store.set_token(username, args.token)
+    token_store.set_active(username)
+    print(f"Authenticated as {username}.")
+
+
+# fmt: off
+
+args_description = [
+    cli.Cmd("to|ken", None, "manage access tokens", [
```

Comment:
I like that you added a shortened version of token here, adds a clearer polish to this new set of subcommands!